### ISSUE TITLE
`PROXY_UNREACHABLE_ERROR` is hard to catch when executed via command

### ISSUE THREAD

**Issue by lukadschaak** [FriendsOfSymfony/FOSHttpCache#574]
I have a question regarding your idea of handling errors, caused by the proxy.

In my special case i execute migration via `doctrine:migrations:migrate`. Some of the migrations create objects, that trigger a purge of the cache system. In our case that is Varnish.

Our problem currently: we are excution the migrations inside a deployment step, where our varnish is not yet reachable.

Technically there are requests written to the queue and inside the `console.terminate` event, `flush` ist called. The `HttpDispatcher` creates exceptions, that are handled by the `CacheInvalidator` with dispatching events. After that, the exceptions are thrown again. I tried to use Symfony's `console.error` event, but it seems, that after `console.terminate`, `console.error` is not dispatched again.

In short: I am not able to catch that error. I will try to build a workaround, that our custom purge listener are not called when commands are executed. Bute i would be interessted if have any idea for a more flexible solution in this package.

**Comment by dbu** [comment]
hm. do you purge the whole varnish cache during the deployment (resp start a new instance of varnish)? if not, don't you have a problem with obsolete cache entries if invalidation fails?

if the cache is purged in a different way, my recommended way would be to use a different configuration during deployment that uses the Noop proxy client instead of the Varnish proxy client. (if you use symfony, that would be an environment e.g. `deploy` with a different configuration for FOSHttpCache and using APP_ENV=deploy when running the migrations). 

if that is no option, you could [decorate](https://symfony.com/doc/current/service_container/service_decoration.html) the `fos_http_cache.cache_manager` service and swallow the exception (but that would hide invalidation failures in other contexts than deployment as well)

**Comment by lukadschaak** [comment]
Thanks for you answer.

First: Yes we purge the varnish cache on deployment by restarting the instance. So no problem with obsolete chaches here.

Using a different environment is kind of tricky, because that would mean a complete different configuration for database, external services and so on. Of course, most of that can simply be extended, but it seems to complex for me to use a whole new environment just for that workaround. But, you are right. That would be an option.

I guess the decoration would work also good, but this is my least favorit solution.

The solution we have now come up with is a simple command listener. This looks at which command we are currently executing and then disables the cache. To do this, we had to modify our own cache listeners so that they don't even try to talk to Varnish if they're not supposed to.

```
class DeactivateCacheOnMigrateCommandListener implements EventSubscriberInterface
{
    public function __construct(
        private readonly CacheActivator $cacheActivator,
    ) {
    }

    public static function getSubscribedEvents(): array
    {
        return [
            ConsoleEvents::COMMAND => 'onConsoleCommand',
        ];
    }

    public function onConsoleCommand(ConsoleCommandEvent $event): void
    {
        if ('doctrine:migrations:migrate' === $event->getCommand()->getName()) {
            $this->cacheActivator->deactivateCaching();
        }
    }
}
```

Thanks for your detailed answer anyway. Maybe this will help someone in the future :)

**Comment by dbu** [comment]
that solution looks sane to me.

if you had many different listeners that invalidate cache, it could make sense to decorate the FOSHttpCache client with a wrapper that provides `deactivate` and only forwards to the inner class if its not deactivated.
